src/lib/preferences.ts

```diff
@@ -201,3 +215,29 @@ export function watchSelectedThemes(cb) {
         },
     };
 }
+
+/**
+ * A custom React hook that returns a state variable and a function that can be
+ * used to set both the state and the preference at the same time.
+ *
+ * @param key a preference (as one would use in `preferences.get(key)`)
+ */
+export function usePreference<KeyT extends ValidPreference>(
+    key: KeyT,
+): [PreferencesType[KeyT], (v: PreferencesType[KeyT]) => void] {
+    const [value, stateSetter] = React.useState(get(key));
+
+    const setStateAndPreference = (v: PreferencesType[KeyT]) => {
+        stateSetter(v);
+        set(key, v);
+    };
+
+    // TODO: maybe add a watcher here for if a different component updates the value?
+    // Something like...
+    //
+    // React.useEffect(() => {
```

Comment:
I think this would be great. I had just implemented a `useUser` earlier today which does the watch thing and it's great. I think I'll probably ditch that and do a `useData` based off of this though as a more generic way of doing things.

Reply:
Awesome, `useUser`/`useData` sounds really useful too.  FWIW I think a dedicated `useUser` function could be nice even if it just wraps the data function, since it's used everywhere and sort of special as data goes.

One notable difference is that `data.get("user")` is always defined(?) whereas most other data entries can return `undefined`.  If we eventually turn `strictNullChecks` on, it will be much more ergonomic to have a user function that guarantees a non-null return value.